From 666ac931cf38f1114b0dd92f3040b32a2f715d60 Mon Sep 17 00:00:00 2001
From: "wenbiao.zhang" <wenbiao.zhang@amlogic.com>
Date: Mon, 4 Aug 2014 14:28:59 +0800
Subject: [PATCH 4645/5965] PD #95128: init deep_suspend_flag and clear
 interrupt before deep sleep to avoid auto resume for the first time suspend

---
 kernel/power/suspend.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/kernel/power/suspend.c b/kernel/power/suspend.c
index 05b28657a746..d1222cd3eaf3 100755
--- a/kernel/power/suspend.c
+++ b/kernel/power/suspend.c
@@ -134,7 +134,7 @@ static int suspend_test(int level)
  * freeze processes.
  */
  
-int deep_suspend_flag;
+int deep_suspend_flag=0;
 
 static int suspend_prepare(suspend_state_t state)
 {
@@ -150,6 +150,11 @@ static int suspend_prepare(suspend_state_t state)
 		goto Finish;
 
 	deep_suspend_flag=1;
+	#ifdef CONFIG_AML_GPIO_KEY
+	extern void clr_pwr_key(void);
+	clr_pwr_key();
+	#endif
+    
 	error = suspend_freeze_processes();
 	if (!error)
 		return 0;
-- 
2.19.0

